packages/button/src/ButtonContent/ButtonContent.tsx

```diff
@@ -44,40 +55,68 @@ export const ButtonContent = ({
   const isIconOnlyButton = ((leftGlyph || rightGlyph) && !children) ?? false;
   const iconProps = { variant, size, darkMode, disabled, isIconOnlyButton };
 
-  return (
-    <>
-      {/* Ripple cannot wrap children, otherwise components that rely on children to render dropdowns will not be rendered due to the overflow:hidden rule. */}
-      <div className={cx(rippleStyle)} ref={rippleRef} />
+  const DefaultContent = ({
+    className,
+  }: Pick<ButtonContentProps, 'className'>) => (
+    <div
+      className={cx(
+        buttonContentStyle,
+        buttonContentSizeStyle[size],
+        {
+          [darkModeRightGlyphStyles]: !!rightGlyph && darkMode,
+        },
+        className,
+      )}
+    >
+      {leftGlyph && (
+        <ButtonIcon
+          glyph={leftGlyph}
+          className={leftGlyphStyles}
+          {...iconProps}
+        />
+      )}
 
-      <div
-        className={cx(buttonContentStyle, buttonContentSizeStyle[size], {
-          [css`
-            justify-content: space-between;
-          `]: !!rightGlyph && darkMode,
-        })}
-      >
-        {leftGlyph && (
-          <ButtonIcon
-            glyph={leftGlyph}
-            className={css`
-              justify-self: right;
-            `}
-            {...iconProps}
-          />
-        )}
+      {children}
 
-        {children}
+      {rightGlyph && (
+        <ButtonIcon
+          glyph={rightGlyph}
+          className={rightGlyphStyles}
+          {...iconProps}
+        />
+      )}
+    </div>
+  );
 
-        {rightGlyph && (
-          <ButtonIcon
-            glyph={rightGlyph}
-            className={css`
-              justify-self: left;
-            `}
-            {...iconProps}
+  if (isLoading) {
+    return (
+      <>
+        <div className={rippleStyle} ref={rippleRef} />
```

Comment:
i don't think we need this, since we probably don't need the ripple when loading